apps/editors/tests/test_review_scenarios.py

```diff
@@ -0,0 +1,134 @@
+"""Real life review scenarios.
+
+For different add-on and file statuses, test reviewing them, and make sure then
+end up in the correct state.
+"""
+import pytest
+
+import amo
+import amo.tests
+from addons.models import Addon
+from editors import helpers
+from files.models import File
+from versions.models import Version
+
+
+@pytest.fixture
+def mock_request(rf, db):  # rf is a RequestFactory provided by pytest-django.
+    request = rf.get('/')
+    request.user = request.amo_user = amo.tests.user_factory()
+    return request
+
+
+@pytest.fixture
+def addon_with_files(db):
+    """Return an add-on with one version and four files.
+
+    By default the add-on is public, and the files are: beta, disabled,
+    unreviewed, unreviewed.
+    """
+    addon = Addon.objects.create()
+    version = Version.objects.create(addon=addon)
+    for status in [amo.STATUS_BETA, amo.STATUS_DISABLED, amo.STATUS_UNREVIEWED,
+                   amo.STATUS_UNREVIEWED]:
+        File.objects.create(version=version, status=status)
+    return addon
+
+
+@pytest.mark.parametrize(
+    'review_action,addon_status,file_status,review_class,review_type,'
+    'final_addon_status,final_file_status',
```

Comment:
Those are the "fixtures" that will be automatically provided by this parametrization.

The `test_review_scenario` will be run once for each scenario (once for each tuple), and each of its parameters will take a value (in order) from the scenario tuple.